### DISCUSSION TITLE
2 NVIDIA GPU's with one monitor connected to each of them, i3wm, xrandr

### DISCUSSION THREAD

**Discussion by YAUIO** [i3/i3#6024] [Show and tell]
Hi, was dealing with setting up a gtx 1070ti and gtx 1060 with two monitors. One is connected to the first GPU, second to the second. I was searching for a solution and only xinerama was a working one, while xrandr with xorg.conf done by nvidia-settings got second monitor to the black screen with an X cursor. ([example of a thread without a solution](https://www.reddit.com/r/i3wm/comments/aqmb1z/2_nvidia_gpus_3_monitors_i3wm_to_rule_them_all/)) SOO i just wanted to share the solution, and maybe it will help somebody some day.

1. You don't need to change anything in your i3conf
2. You should setup your xorg.conf for your primary screen and first GPU ONLY. Getting second ones here won't work and will break everything
My conf:
```
# nvidia-settings: X configuration file generated by nvidia-settings
# nvidia-settings:  version 550.67

Section "ServerLayout"
    Identifier     "Layout0"
    Screen      0  "Screen0" 0 0
    InputDevice    "Keyboard0" "CoreKeyboard"
    InputDevice    "Mouse0" "CorePointer"
    Option         "Xinerama" "0"
EndSection

Section "Files"
EndSection

Section "InputDevice"
    # generated from default
    Identifier     "Mouse0"
    Driver         "mouse"
    Option         "Protocol" "auto"
    Option         "Device" "/dev/psaux"
    Option         "Emulate3Buttons" "no"
    Option         "ZAxisMapping" "4 5"
EndSection

Section "InputDevice"
    # generated from default
    Identifier     "Keyboard0"
    Driver         "kbd"
EndSection

Section "Monitor"
    # HorizSync source: edid, VertRefresh source: edid
    Identifier     "Monitor0"
    VendorName     "Unknown"
    ModelName      "AOC AG251FWG2"
    HorizSync       255.0 - 255.0
    VertRefresh     48.0 - 242.0
    Option         "DPMS"
EndSection

Section "Device"
    Identifier     "Device0"
    Driver         "nvidia"
    VendorName     "NVIDIA Corporation"
    BoardName      "NVIDIA GeForce GTX 1070 Ti"
    BusID          "PCI:8:0:0"
EndSection

Section "Screen"
    Identifier     "Screen0"
    Device         "Device0"
    Monitor        "Monitor0"
    DefaultDepth    24
    Option         "Stereo" "0"
    Option         "nvidiaXineramaInfoOrder" "DP-0"
    Option         "metamodes" "1920x1080_240 +0+0"
    Option         "SLI" "Off"
    Option         "MultiGPU" "Off"
    Option         "BaseMosaic" "off"
    SubSection     "Display"
        Depth       24
    EndSubSection
EndSection
```
4. You NEED to turn off [DRM kernel mode setting](https://wiki.archlinux.org/title/NVIDIA#DRM_kernel_mode_setting)
5. Finally, xrandr should work fine with two GPU's and you need just to configure it as usual
_if it doesn't work try xrandr --setprovideroutputsource <primary> <secondary> from xrandr --listproviders _

Hope this helps!

**Comment by orestisfl** [comment]
Thanks for sharing, that's probably useful for anyone using X11.

Just curious, what's the motivation for doing this?

**Reply by YAUIO** [reply]
For me it was just the fact that on my new 1070ti dvi-d didn't work and i didn't want to wait for an dvi-hdmi adapter to come. So i decided to just get myself a challenge alongside with trying something new and getting second monitor to work

_so basically i just love to solve some linux problems_
